types/react/index.d.ts

```diff
@@ -2282,15 +2255,28 @@ declare namespace React {
     }
 }
 
+type Defaultize<Props, Defaults> =
+    & Pick<Props, Exclude<keyof Props, keyof Defaults>>
+    & Partial<Pick<Props, Extract<keyof Props, keyof Defaults>>>
+    & Partial<Defaults>;
+
 declare global {
     namespace JSX {
         // tslint:disable-next-line:no-empty-interface
         interface Element extends React.ReactElement<any> { }
         interface ElementClass extends React.Component<any> {
             render(): React.ReactNode;
         }
+        // not sure this is needed anymore
```

Comment:
They are technically still used, though if removed, the compiler can still operate without them. If `ElementAttributesProperty` isn't present, the type of the first parameter of the element factory (class or sfc) will be used instead. Today, that's actually _slightly_ different for a react consumer, since it doesn't usually include `children`.